src/Stratis.Bitcoin.Features.SignalR/Broadcasters/BroadcasterBase.cs

```diff
@@ -36,16 +39,52 @@ public void Init(ClientEventBroadcasterSettings broadcasterSettings)
                 $"Broadcast {this.GetType().Name}",
                 async token =>
                 {
-                    foreach (IClientEvent clientEvent in this.GetMessages())
+                    using (var cancellationTokenSource = new CancellationTokenSource())
```

Comment:
So the idea here is to create a linked token from `this.nodeLifetime` and then set the `CancelAfter`. So there will only be one using:

```
using (var linkedTokenSource = CancellationTokenSource.CreateLinkedTokenSource(this.nodeLifetime.ApplicationStopping, token))
{
    linkedTokenSource.CancelAfter(new TimeSpan(0, 0, 10));
    ....
```

Then pass that to `GetMessages`

Reply:
Will do

Reply:
Done